enhancements/sdk-fbc-run-bundle.md

```diff
@@ -0,0 +1,279 @@
+
+## Motivation
+
+One of the main goals of the Operator SDK is to enable operator developers to install and run their operators using Operator Lifecycle Manager (OLM). Currently, there are two commands in the Operator SDK for installing an operator bundle (`run bundle`) and upgrading an operator bundle to a newer version (`run bundle-upgrade`) using OLM. Operator SDK installs and upgrades the operator bundles using opm tooling that inherently uses a SQLite database under the hood, which is responsible for generating the registry database as well as the index images. We use `opm registry add` under the hood to create an ephemeral index image during the `run bundle(-upgrade)` workflow; however, that command is being deprecated.
+
+As OLM is transitioning to declarative config, which is a more human readable and consumable format of a catalog that is in the form of `JSON/YAML`, Operator SDK should transition the existing `run bundle(-upgrade)` commands to support declarative config index images and default to using File-Based Catalogs so that Operator authors can continue using Operator SDK to install/upgrade their operator bundles using the new FBC format.
+
+This proposal discusses changes to the `run bundle` functionality to be able to support File-Based Catalogs and how the registry pod should be created to support the generated FBC format i.e., to modify/replace the existing opm commands used to create the ephemeral index image during registry pod creation by leveraging the Operator Registry APIs.
+
+### Goals
+
+- Operator SDK’s `run bundle` command should work with File-Based Catalogs and be able to install an operator bundle using a valid FBC.
+
+- Operator SDK should insert a bundle image into an existing index image that gets passed in as a parameter in the `run bundle` command.
+
+- Operator SDK's `run bundle` command should still be able support SQLite index images along with the FBC images until `opm registry add` and `opm index` and other SQLite-related commands are removed in a future OpenShift release.
+
+- The CLI for `run bundle` command should remain exactly the same i.e. no user-facing changes.
+
+### Non-Goals
+
+- Operator SDK’s `run bundle` and `run bundle-upgrade` commands share a lot of common functionality and this EP does not discuss the impact on the run bundle-upgrade command. A separate EP is required for the `run bundle-upgrade` command.
+
+- Migrating from old catalog images to newer catalog versions with FBC support is not a goal of this EP.
+
+- Another non-goal, is that `run bundle` is **NOT** intended to help users build and manage File-Based indexes. It is an opinionated command to help users test simple bundle-based execution scenarios with ephemeral indexes.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an Operator SDK user, I want to be able to install an operator bundle with File-Based Catalog support using OLM without having to provide a catalog.
```

Comment:
```suggestion
As an SDK/OLM maintainer, I want to ensure the SDK is updated to use the new format and APIs provided by OLM/OPM so that, we can sort out the technical debts and be able to no longer support the old syntax and format when they are no longer supported by OPM/OLM.  
```